plutus-core/plutus-core/src/PlutusCore/MkPlc.hs

```diff
@@ -229,6 +243,17 @@ mkIterLamAbs
 mkIterLamAbs args body =
     foldr (\(VarDecl ann name ty) acc -> lamAbs ann name ty acc) body args
 
+-- | Bind a list of variables via a product.
+mkProdLamAbs
+    :: TermLike term tyname name uni fun
+    => ann
+    -> name
+    -> [VarDecl tyname name uni fun ann]
+    -> term ann
+    -> term ann
+mkProdLamAbs ann name args body =
+    lamAbs ann name (TyProd ann (fmap _varDeclType args)) $ foldr (\(vd, i) acc -> termLet ann (Def vd (proj ann i (var ann name))) acc) body (zip args [0..])
```

Comment:
I don't think it would make much difference. In any case, they get turned into let-bindings and can get moved around quite a bit before they get turned into labmdas.